src/test/java/apoc/export/json/ExportJsonTest.java

```diff
@@ -0,0 +1,323 @@
+package apoc.export.json;
+
+import apoc.export.util.JsonResults;
+import apoc.graph.Graphs;
+import apoc.util.TestUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.test.TestGraphDatabaseFactory;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Scanner;
+
+import static apoc.util.MapUtil.map;
+import static org.junit.Assert.assertEquals;
+
+public class ExportJsonTest {
+
+    private static GraphDatabaseService db;
+    private static File directory = new File("target/import");
+
+    static { //noinspection ResultOfMethodCallIgnored
+        directory.mkdirs();
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        db = new TestGraphDatabaseFactory()
+                .newImpermanentDatabaseBuilder()
+                .setConfig(GraphDatabaseSettings.load_csv_file_url_root, directory.getAbsolutePath())
+                .setConfig("apoc.export.file.enabled", "true")
+                .newGraphDatabase();
+        TestUtil.registerProcedure(db, ExportJson.class, Graphs.class);
+        db.execute("CREATE (f:User {name:'foo',age:42,male:true,kids:['a','b','c'], born:localdatetime('2015185T19:32:24'), place:point({latitude: 13.1, longitude: 33.46789})})-[:KNOWS {since: 1993}]->(b:User {name:'bar',age:42}),(c:User {age:12})").close();
+    }
+
+    @After
+    public void tearDown() {
+        db.shutdown();
+    }
+
+    @Test
+    public void testExportAllJson() throws Exception {
+        String filename = "all.json";
+        File output = new File(directory, filename);
+        TestUtil.testCall(db, "CALL apoc.export.json.all({file},null)", map("file", output.getAbsolutePath()),
+                (r) -> {
+                    assertResults(output, r, "database");
+                }
+        );
+        assertEquals(JsonResults.expectedAllGraphData(), new Scanner(output).useDelimiter("\\Z").next());
+    }
+
+    @Test
+    public void testExportPointMapDatetimeJson() throws Exception {
+        String filename = "mapPointDatetime.json";
+        File output = new File(directory, filename);
+        String query = "return {a: 1, b: {c: 1, d:'a', e: {f: [1,3,5]}}} as map, " +
+                "datetime('2015-06-24T12:50:35.556+0100') AS theDateTime, " +
+                "localdatetime('2015185T19:32:24') AS theLocalDateTime," +
+                "point({latitude: 13.1, longitude: 33.46789}) as point," +
+                "date('+2015-W13-4') as date," +
+                "time('125035.556+0100') as time," +
+                "localTime('12:50:35.556') as localTime";
+        TestUtil.testCall(db, "CALL apoc.export.json.query({query},{file})", map("file", output.getAbsolutePath(),"query",query),
+                (r) -> {
+                    assertEquals(true,r.get("source").toString().contains("kernelTransaction: cols(7)"));
```

Comment:
No
Fare attenzione ai copia/incolla:
```suggestion
                    assertTrue(r.get("source").toString().contains("kernelTransaction: cols(7)"));
```
**vedere in tutti gli altri punti del codice facendo molta attenzione a non perdersi nulla per strada**

Bisogna fare attenzione perché è anche una cosa che suggerisce Intellij, fare sempre attezione ai messaggi dell'IDE